### ISSUE TITLE
AnnotationWithQuickFixesHoverTest fails when run from eclipse 

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#1538]
AnnotationWithQuickFixesHoverTest is flaky

**Comment by cdietrich** [comment]
[org.eclipse.xtext.ui.tests.editor.hover.AnnotationWithQuickFixesHoverTest.txt](https://github.com/eclipse/xtext-eclipse/files/5244948/org.eclipse.xtext.ui.tests.editor.hover.AnnotationWithQuickFixesHoverTest.txt)

**Comment by cdietrich** [comment]
for some reason editor has no focus. i have no idea why
@szarnekow any hint?

**Comment by cdietrich** [comment]
adding

```
Display current = Display.getCurrent();
		if (current != null) {
			while (current.readAndDispatch()) {
				// process queued ui events
			}
		}
```

to setup before superCall seems to help.
but why
and shouldnt this be done by PDE?

**Comment by cdietrich** [comment]
hmmm ok

```
	protected static void closeWelcomePage() throws InterruptedException {
		IIntroManager introManager = getWorkbench().getIntroManager();
		IIntroPart intro = introManager.getIntro();
		if (intro != null) {
			introManager.closeIntro(intro);
			Display current = Display.getCurrent();
			if (current != null) {
				while (current.readAndDispatch()) {
					// process queued ui events
				}
			}
		}
	}
```

**Comment by cdietrich** [comment]
@iloveeclipse any idea why this is needed?

**Comment by cdietrich** [comment]
also wonder if we should call
waitForEventProcessing
before and after the setup & teardown block
@szarnekow what do you think?

**Comment by iloveeclipse** [comment]
Why the welcome page is configured to be shown? I think one could disable this via preference.

**Comment by cdietrich** [comment]
we just use the tycho surefire / pde default

**Comment by szarnekow** [comment]
> also wonder if we should call
> waitForEventProcessing
> before and after the setup & teardown block
> @szarnekow what do you think?

Yes, we should.

**Comment by cdietrich** [comment]
Fixed in 2.24
